### PR TITLE
feat: add feature processor APIs to public doc

### PR DESCRIPTION
**Pull request by can-sun**
*Issue #, if available:*

Adding feature processor APIs to public doc.

*Description of changes:*

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [X] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [X] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [X] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [X] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [X] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [X] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [X] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [X] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [X] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by alex-tang — APPROVED** [review]

File: doc/requirements.txt [hunk 1]
```diff
@@ -4,3 +4,5 @@ docutils==0.15.2
 packaging==20.9
 jinja2<3.1
 schema==0.7.5
+pyspark==3.3.1
+sagemaker-feature-store-pyspark-3.3
```
Comment (alex-tang) [inline doc/requirements.txt:8]: Is this necessary? These dependencies are specified in the extras. I don't see other ones here https://github.com/aws/sagemaker-python-sdk/tree/master/requirements/extras.

**Review by alex-tang — APPROVED** [review]

**Review by akrishna1995 — COMMENTED** [review]
/bot run all

**Comment by akrishna1995** [comment]
Can you confirm if black-check and flake8 is passing in local ?

**Review by akrishna1995 — APPROVED** [review]
Thanks for fixing the formatting

**Comment by akrishna1995** [comment]
Confirmed with Can Sun that the failure is unrelated to change

**Pull request merged**